dbms/src/Storages/S3/CheckpointManifestS3Set.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Storages/S3/CheckpointManifestS3Set.h>
+#include <Storages/S3/S3Filename.h>
+
+namespace DB::S3
+{
+CheckpointManifestS3Set
+CheckpointManifestS3Set::getFromS3(const S3::TiFlashS3Client & client, StoreID store_id)
+{
+    const auto store_prefix = S3::S3Filename::fromStoreId(store_id).toManifestPrefix();
+
+    std::vector<CheckpointManifestS3Object> manifests;
+
+    listPrefix(client, client.bucket(), store_prefix, [&](const Aws::S3::Model::ListObjectsV2Result & result) {
```

Comment:
We need some metrics about how many times S3 API has been called.

Reply:
Need to wait for this PR https://github.com/pingcap/tiflash/pull/6913/files#diff-a9609b7aa8c2b0399aeccb4cd74379bceaea166374dd71247e81f7ba50671a36R286